### PR TITLE
[EPMCDLAB-1617]: Created REST service for managing users' roles

### PR DESCRIPTION
**Pull request by ochaparin**

### CODE REVIEW

File: services/self-service/src/main/java/com/epam/dlab/backendapi/dao/UserRoleDao.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import static com.mongodb.client.model.Filters.eq;
+
+public class UserRoleDao extends BaseDAO {
+
+	public List<UserRoleDto> getUserRoles() {
+		return find(MongoCollections.ROLES, UserRoleDto.class);
+	}
+
+	public void createOrUpdate(UserRoleDto dto) {
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/dao/UserRoleDao.java]: please separate create and update

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/UserRolesResource.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response getRoles(@Auth UserInfo userInfo) {
+		log.debug("Getting all roles for admin {}...", userInfo.getName());
+		return Response.ok(userRolesService.getUserRoles()).build();
+	}
+
+	@PUT
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response upsertRole(@Auth UserInfo userInfo, UserRoleDto dto) {
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/UserRolesResource.java]: separate create and update

File: services/self-service/src/main/java/com/epam/dlab/backendapi/service/impl/UserRolesServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	@Override
+	public void createOrUpdateRole(UserRoleDto dto) {
+		userRoleDao.createOrUpdate(dto);
+	}
+
+	@Override
+	public void removeRoleIfExists(String roleId, String user) {
+		if (userRoleDao.roleExists(roleId)) {
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/service/impl/UserRolesServiceImpl.java]: why do we need this checking? Removing role without cheking should work

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/UserRolesResource.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Slf4j
+@Path("roles")
+@RolesAllowed("roles")
+public class UserRolesResource {
+
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/UserRolesResource.java]: add method for assigning multiple roles for group/user

File: services/self-service/src/main/java/com/epam/dlab/backendapi/dao/UserRoleDao.java (hunk 1 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/dao/UserRoleDao.java]: Done

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/UserRolesResource.java (hunk 2 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/UserRolesResource.java]: Done

File: services/self-service/src/main/java/com/epam/dlab/backendapi/service/impl/UserRolesServiceImpl.java (hunk 3 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/service/impl/UserRolesServiceImpl.java]: Updated

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/UserRolesResource.java (hunk 4 above)
Comment (ochaparin) [reply services/self-service/src/main/java/com/epam/dlab/backendapi/resources/UserRolesResource.java]: Added

**Pull request merged**